utils/ssh.py

```diff
@@ -292,47 +310,98 @@ def cpu_spike(self, seconds=60, cpus=2, **kwargs):
+            logger.info('For this purpose, temporary file name is %r', tempfilename)
             scp = SCPClient(self.get_transport(), progress=self._progress_callback).put(
                 local_file, tempfilename, **kwargs)
             self.run_command('mv {} {}'.format(tempfilename, remote_file))
             return scp
+        elif self.is_pod:
+            tmp_folder_name = 'automation-{}'.format(fauxfactory.gen_alpha().lower())
+            logger.info('For this purpose, temporary folder name is /tmp/%s', tmp_folder_name)
+            # Clean up container's temporary folder
+            self.run_command('rm -rf /tmp/{0}'.format(tmp_folder_name))
+            # Create/Clean up the host's temporary folder
+            self.run_command(
+                'rm -rf /tmp/{0}; mkdir -p /tmp/{0}'.format(tmp_folder_name), ensure_host=True)
+            # Now upload the file to the openshift host
+            tmp_file_name = 'file-{}'.format(fauxfactory.gen_alpha().lower())
+            tmp_full_name = '/tmp/{}/{}'.format(tmp_folder_name, tmp_file_name)
+            scp = SCPClient(self.get_transport(), progress=self._progress_callback).put(
+                local_file, tmp_full_name, **kwargs)
+            # use oc rsync to put the file in the container
+            assert self.run_command(
+                'oc rsync /tmp/{} {}:/tmp/'.format(tmp_folder_name, self._container),
+                ensure_host=True)
+            # Move the file onto correct place
+            assert self.run_command('mv {} {}'.format(tmp_full_name, remote_file))
+            return scp
         else:
             if self.username == 'root':
                 return SCPClient(self.get_transport(), progress=self._progress_callback).put(
                     local_file, remote_file, **kwargs)
             # scp client is not sudo, may not work for non sudo
             tempfilename = '/home/{user_name}/temp_{random_alpha}'.format(
                 user_name=self.username, random_alpha=fauxfactory.gen_alpha())
+            logger.info('For this purpose, temporary file name is %r', tempfilename)
             scp = SCPClient(self.get_transport(), progress=self._progress_callback).put(
                 local_file, tempfilename, **kwargs)
             self.run_command('mv {temp_file} {remote_file}'.format(temp_file=tempfilename,
                                                                    remote_file=remote_file))
             return scp
 
     def get_file(self, remote_file, local_path='', **kwargs):
-        logger.info("Transferring remote file {remote_file} to local {local_path}".format(
-            remote_file=remote_file, local_path=local_path))
+        logger.info("Transferring remote file %r to local %r", remote_file, local_path)
+        base_name = os_path.basename(remote_file)
         if self.is_container:
-            tempfilename = '/share/temp_{}'.format(fauxfactory.gen_alpha())
+            tmp_file_name = 'temp_{}'.format(fauxfactory.gen_alpha())
```

Comment:
it would be good to use more meaning full var name in this case. it is difficult to read code when several variables have almost the same names